src/Network/Minio/API.hs

```diff
@@ -121,8 +123,13 @@ buildRequest ri = do
                    (Map.lookup r awsRegionMap)
               else return $ connectHost ci
 
+  sha256Hash <- if | connectIsSecure ci ->
+                       -- if secure connection
+                       return $ C.pack "UNSIGNED-PAYLOAD"
```

Comment:
Since, `sha256Hash` has type `ByteString`, and we have the `OverloadedStrings` extension on all files by default (through a cabal file setting), the compiler can infer the type of string literals. So it is sufficient to have: `return "UNSIGNED-PAYLOAD"` above. The compiler will perform the rest automatically.

Reply:
Now i see..